lib/src/realm_object.dart

```diff
@@ -206,6 +206,16 @@ class RealmObject {
     RealmAccessor.setDefaults<T>(values);
     return true;
   }
+
+  @override
+  // ignore: hash_and_equals
+  bool operator ==(Object other) {
+    final o = other;
+    if (identical(this, o)) return true;
+    if (o is! RealmObject) return false;
+    if (!isManaged || !o.isManaged) return false;
+    return realmCore.equals(handle, o.handle);
```

Comment:
I would suggest we follow the other places and keep passing realm objects to realm_core and get the handles from inside the realm_core implementation. I will put suggestions on the relevant places

Reply:
Fixed